### PR TITLE
fix: Auto-Suggestion for fields and values for structure LG

### LINKED ISSUE #2680: LG editing: card suggestion should suggest fields and possible values
when editing structure LG, the editor should be able to do auto-suggestions for fields of current struct and possible values.

### PR DESCRIPTION
**Pull request by cosmicshuai**
closes #2680 

Now when user entering structure LG. The editor is able to suggest all possible fields and a value for each field.  The effect can be 

Full LG mode:
full lg card
full lg suggest

Inline LG mode:
inline lg attachment
inline lg card

### CODE REVIEW

**Comment by cosmicshuai** [comment]
@cwhitten 
I did some modification on your original proposal. 
If the editor do the suggestion for fields and values at one time, due to all the content can only be shown on a single line (newline character has no effect in suggestion window), users will see a messy suggestion.
So I changed the suggestions into seperated fileds way. Each filed and its correspongding value will become one suggestion item. And all the fields already existed in current card won't show up. 
After the last field is finished. The editor will do the trimming automatically.

**Comment by cwhitten** [comment]
ok @cosmicshuai - is this still a draft or ready for review?

**Comment by cosmicshuai** [comment]
> ok @cosmicshuai - is this still a draft or ready for review?

It's ready for review.

File: Composer/packages/tools/language-servers/language-generation/src/utils.ts [hunk 1]
```diff
@@ -153,6 +153,20 @@ export const cardTypes = [
...
+  SuggestionActions: 'Text | ${Some_CardAction}',
+  subtitle: 'An Example Subtitle',
+  text: 'Some text',
+  image: 'https://example.com/demo.jpg',
+  buttons: 'Text | ${Some_CardAction}',
+  contenttype: 'adaptivecard',
+  content: '${json(fromFile("../../card.json"))}',
+  name: 'An Example Name',
+};
+
```
Comment (a-b-r-o-w-n) [inline Composer/packages/tools/language-servers/language-generation/src/utils.ts:169]: Can these use the vs code placeholder symantics?

For example: `text: '${1:Some text}'

See the example below of a snippet that does this.
Screen Recording 2020-04-16 at 09 19 AM

Comment (cosmicshuai) [reply Composer/packages/tools/language-servers/language-generation/src/utils.ts:169]: I tried using snippet string with placeholder during adding suggestion of builtin functions. However, the completionItem in LSP cannot do the placeholder directly. I think it needs some transformation in monaco client side. Furthermore, the snippet ${1: some text} looks confusing in the context of we can using expression as a value here.

Comment (a-b-r-o-w-n) [reply Composer/packages/tools/language-servers/language-generation/src/utils.ts:169]: I see. This aren't really "snippets" in the vs code sense.

**Comment by a-b-r-o-w-n** [comment]
This was likely not introduced in this change, but I am seeing the suggestions being clipped:

image

**Comment by cwhitten** [comment]
@a-b-r-o-w-n is the visual editor container doing the clipping?

**Comment by a-b-r-o-w-n** [comment]
@cwhitten possibly. I thought it was the form, but didn't see any element with `overflow: hidden`. I didn't check the visual editor though.

**Pull request merged**